cajita/src/Cajita_GlobalGrid_impl.hpp

```diff
@@ -211,6 +211,74 @@ GlobalGrid<MeshType>::blockRank( const int i, const int j ) const
     return blockRank( cr );
 }
 
+//---------------------------------------------------------------------------//
+// Get the MPI sub communicator for given direction
+template <class MeshType>
+MPI_Comm GlobalGrid<MeshType>::subComm( const int dim ) const
+{
+    MPI_Comm sub_comm;
+
+    // Set colors same within the dimBlockId of the dim
+    MPI_Comm_split( comm(), dimBlockId( dim ), blockId(), &sub_comm );
+    return sub_comm;
+}
+
+//---------------------------------------------------------------------------//
+// Get the MPI sub communicator for reduction in the given direction
+template <class MeshType>
+MPI_Comm GlobalGrid<MeshType>::reductionSubComm( const int dim ) const
```

Comment:
I think this will be easier if you pass 2 dimensions here (and restrict to 3d only) and rename to `subComm`

Reply:
`subComm` is used for creating sub communicator in a given direction and this `reductionSubComm` is used for creating sub communicator when there is reduction in the given direction.

Reply:
But can't `subComm` be used for a reduction into 1d and `reductionSubComm` be used for reduction into 2d? Why not pass the 2 "non-reduction" dimensions here like you do in the case above?